processor/elasticprocessor/processor.go

```diff
@@ -2,46 +2,57 @@ package elasticprocessor
 
 import (
 	"context"
-	"strings"
-
-	"github.com/tommyers-elastic/opentelemetry-collector-contrib/processor/elasticprocessor/internal/hostmetrics"
 
+	"github.com/elastic/opentelemetry-lib/remappers/hostmetrics"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	"go.opentelemetry.io/collector/processor"
-
 	"go.uber.org/zap"
 )
 
+// remapper interface defines the Remap method that should be implemented by different remappers
+type remapper interface {
+	Remap(pmetric.ScopeMetrics, pmetric.MetricSlice, pcommon.Resource)
+}
+
 type ElasticProcessor struct {
-	cfg     *Config
-	logger  *zap.Logger
-	storage map[string]any
+	cfg       *Config
+	logger    *zap.Logger
+	storage   map[string]any
+	remappers []remapper
 }
 
 func newProcessor(set processor.CreateSettings, cfg *Config) *ElasticProcessor {
-	return &ElasticProcessor{cfg: cfg, logger: set.Logger, storage: make(map[string]any)}
+	var remappers []remapper
+	// Initialize the remapper slice if AddSystemMetrics is enabled
+	if cfg.AddSystemMetrics {
+		remappers = []remapper{
+			hostmetrics.NewRemapper(set.Logger, hostmetrics.WithSystemIntegrationDataset(true)),
+		}
+	}
+	return &ElasticProcessor{
+		cfg:       cfg,
+		logger:    set.Logger,
+		storage:   make(map[string]any),
```

Comment:
I am not sure what was the decision on the `storage` logic but if you want to keep it then that logic needs to be added in the `processMetrics` (probably before the for loop).

Reply:
storagr was being used in calculation of metricset.period. Now if that's not happening in the opentelemtery-lib, there is no use. 
@tommyers-elastic: WDYT ?

Reply:
> Now if that's not happening in the opentelemtery-lib, there is no use.

It's not. At least not yet. The PoC https://github.com/elastic/opentelemetry-lib/pull/16 has something but it is still not certain if that will work

Reply:
let's remove it if it's not being used for anything

Reply:
Done